### PR TITLE
[PAY-598] Prepopulate TransactionDetails after purchasing $AUDIO

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Populate the TransactionDetails for the just-completed transaction at the end of the BuyAudio modal so that "Review Transaction" can work without a query to Discovery/Identity

- Changes the TransactionDetails type to be more strict
- Updates transaction fee calculation to use fancy Transaction.getEstimatedFee() method and have a default of 5000 Lamports if that fails (for some reason, swaps sometimes don't have estimates?)
- Updates usage of `u64` to `BN` where possible
- Updates the BuyAudio modal success page to use the transaction details to get the AUDIO change and new balance

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

- Uses estimated USD from the modal for now as the "usd" amount of the transaction.
- Also uses the client dayjs date for the prepopulation of the TransactionDetails, but should take note in identity to ignore that and use a `created_at` instead. (`blocktime` doesn't seem to be available right away so this avoids the 'undefined' case - also since this doesn't have to be tied to one particular userbank tx this seems ok)

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Tested E2E using phantom to mock Coinbase Pay

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

Only entry point is currently behind the BUY_AUDIO_ENABLED feature flag

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-transaction-details-modal-wired

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-transaction-details-modal-wired

File: packages/common/src/store/ui/transaction-details/types.ts [hunk 1]
```diff
@@ -24,19 +27,42 @@ export type InAppAudioPurchaseMetadata = {
...
-export type TransactionDetails = {
-  signature: string
-  transactionType: TransactionType
-  method: TransactionMethod
-  date: string
-  change: StringAudio
-  balance: StringAudio
-  metadata: InAppAudioPurchaseMetadata | undefined
-}
+export type TransactionDetails =
```
Comment (piazzatron) [inline packages/common/src/store/ui/transaction-details/types.ts:36]: niceeee

File: packages/web/src/services/audius-backend/BuyAudio.ts [hunk 2]
```diff
@@ -38,14 +38,14 @@ const delay = (ms: number) =>
 
 export const getRootSolanaAccount = async () => {
   await waitForLibsInit()
-  return libs().solanaWeb3Manager.solanaWeb3.Keypair.fromSeed(
-    libs().Account.hedgehog.wallet.getPrivateKey()
-  ) as Keypair
+  return libs().solanaWeb3Manager!.solanaWeb3.Keypair.fromSeed(
```
Comment (piazzatron) [inline packages/web/src/services/audius-backend/BuyAudio.ts:41]: Howcome we need `!`?

File: packages/web/src/services/audius-backend/BuyAudio.ts [hunk 3]
```diff
@@ -242,3 +242,10 @@ export const createTransferToUserBankTransaction = async ({
   tx.add(transferInstruction)
   return tx
 }
+
+export const saveUserBankTransactionMetadata = async (
+  data: InAppAudioPurchaseMetadata
+) => {
+  await waitForLibsInit()
+  // return await libs().identityService!.saveUserBankTransactionMetadata(data)
```
Comment (piazzatron) [inline packages/web/src/services/audius-backend/BuyAudio.ts:250]: intentional comment?

**Review by piazzatron — APPROVED** [review]

File: packages/web/src/services/audius-backend/BuyAudio.ts (hunk 3 above)
Comment (piazzatron) [reply packages/web/src/services/audius-backend/BuyAudio.ts:250]: ah just saw the commit msg

File: packages/web/src/services/audius-backend/BuyAudio.ts (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/services/audius-backend/BuyAudio.ts:41]: all the services are nullable :(

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-transaction-details-modal-wired

**Pull request merged**